src/NonCumulative/Soundness/Properties/LogRel.agda

```diff
@@ -86,4 +98,712 @@ Glu-wellfounded-≡-GluΛ {j} {k} = implicit-extensionality fext (fext (λ l<k 
            Γ ⊢ t ∶ T ®[ i ] a ∈El A≈B →
            ---------------------------
            Γ ⊢ T ∶[ 1 + i ] Se i
-®El⇒ty A≈B t∼a = ®⇒ty A≈B (®El⇒® A≈B t∼a)
\ No newline at end of file
+®El⇒ty A≈B t∼a = ®⇒ty A≈B (®El⇒® A≈B t∼a)
+
+private
+  s:IT®⇒⊢s : ∀ { j k b IT σ s} → 
+            (jA : A ≈ A′ ∈ PERDef.𝕌 j (λ l<j → 𝕌-wellfounded (max j k) (ΠI≤ refl l<j))) → 
+            Glu.⟦ j , (λ l<j → Glu-wellfounded (max j k) (ΠI≤ refl l<j)) , (λ l<j → 𝕌-wellfounded (max j k) (ΠI≤ refl l<j)) ⟧ Δ ⊢ s ∶ sub IT σ ® b ∈El jA →
+            Δ ⊢ s ∶[ j ] sub IT σ
+  s:IT®⇒⊢s {j = j} {k = k} jA tr rewrite 𝕌-wf-gen {max j k} j (λ l<k → ΠI≤ refl l<k) | Glu-wellfounded-≡-GluΛI {j} {k} = ®El⇒tm jA tr
+
+-- ®El respects term equivalence.
+®El-resp-≈ : ∀ {i} (A≈B : A ≈ B ∈ 𝕌 i) →
+             Γ ⊢ t ∶ T ®[ i ] a ∈El A≈B →
+             Γ ⊢ t ≈ t′ ∶[ i ] T →
+             ----------------------------
+             Γ ⊢ t′ ∶ T ®[ i ] a ∈El A≈B
+®El-resp-≈ (ne C≈C j≡1+i j′≡1+i) (ne c≈c′ refl _ , glu) t≈t′ = ne c≈c′ refl refl , record 
+  { t∶T = proj₁ (proj₂ (proj₂ (presup-≈ t≈t′)))
+  ; ⊢T = ⊢T
+  ; krip = λ ⊢σ → proj₁ (krip ⊢σ) , ≈-trans ([]-cong (≈-sym t≈t′) (s-≈-refl (⊢w⇒⊢s ⊢σ))) (proj₂ (krip ⊢σ)) 
+  }
+  where open GluNe glu
+®El-resp-≈ N′ (t® , T≈N) t≈t′ = ®Nat-resp-≈ t® (≈-conv t≈t′ T≈N) , T≈N
+®El-resp-≈ (U {j} refl refl) record { t∶T = t∶T ; T≈ = T≈ ; A∈𝕌 = A∈𝕌 ; rel = rel } t≈t′ 
+  rewrite Glu-wellfounded-≡-GluU {j} rewrite 𝕌-wf-gen j (λ l<j → <-trans l<j (s≤s (≤-reflexive refl))) = record
+    { t∶T = proj₁ (proj₂ (proj₂ (presup-≈ t≈t′))) 
+    ; T≈ = T≈ 
+    ; A∈𝕌 = A∈𝕌 
+    ; rel = ®-resp-≈ A∈𝕌 rel (≈-conv t≈t′ T≈) 
+    }
+®El-resp-≈ (Π {j = j} {k = k} refl jA RT refl refl) record { t∶T = t∶T ; a∈El = a∈El ; IT = IT ; OT = OT ; ⊢IT = ⊢IT ; ⊢OT = ⊢OT ; T≈ = T≈ ; krip = krip } t≈t′    
+  rewrite Glu-wellfounded-≡-GluΛO {j} {k} = record 
+  { t∶T = proj₁ (proj₂ (proj₂ (presup-≈ t≈t′)))
+  ; a∈El = a∈El
+  ; IT = IT
+  ; OT = OT
+  ; ⊢IT = ⊢IT
+  ; ⊢OT = ⊢OT
+  ; T≈ = T≈
+  ; krip = λ ⊢σ → let open ΛRel (krip ⊢σ) in record 
```

Comment:
consider using a lemma in a where clause. let expressions are expanded during type checking